src/pudl/extract/phmsagas.py

```diff
@@ -35,18 +36,21 @@ def process_final_page(self, df, page):
         older years, filter by the list of columns specified for the page, with a
         warning.
         """
-        to_drop = [
-            c
-            for c in df.columns
-            if c not in self._metadata.get_all_columns(page)
-            and c not in self.cols_added
-        ]
-        if to_drop:
-            logger.warning(
-                f"Dropping columns {to_drop} that are not mapped to this asset."
-            )
-            df = df.drop(columns=to_drop, errors="ignore")
-        return df
+        if int(partition["year"]) < 2010:
```

Comment:
This is also only true for transmission and not distribution data, so we want to filter by `form` as well. We should update the docstring to clarify. Otherwise this is super helpful and catches some columns I've missed in earlier mappings.

Reply:
this is a great suggestion! will do for sure

Reply:
@e-belfer is it only `page != "yearly_distribution"` that we should avoid for this?

i was going to add `and "_transmission_" in page` here but `yearly_inspections_and_assessments` is a part of the transmission zip but it just doesn't look like it is included in old years.

Reply:
could also look it up in the page->form map

Reply:
I think that table should probably get renamed to include transmission in it, honestly.

Reply:
But the way you've done it is more robust